packages/next-server/lib/router/router.js

```diff
@@ -364,8 +364,8 @@ export default class Router {
 
   urlIsNew (asPath) {
     const { pathname, query } = parse(asPath, true)
-    const { pathname: curPathname } = parse(this.asPath, true)
-    return curPathname !== pathname || !shallowEquals(query, this.query)
+    const { pathname: curPathname, query: curQuery } = parse(this.asPath, true)
+    return curPathname !== pathname || !shallowEquals(query, curQuery)
```

Comment:
Couldn't the [`search`](https://nodejs.org/docs/latest/api/url.html#url_urlobject_search) property be used here instead of the `query` object? That way we wouldn't have to call `shallowEquals` and could just compare the strings for equality.